lib/backend/etcdv3/etcdv3.go

```diff
@@ -55,13 +57,40 @@ func NewEtcdV3Client(config *apiconfig.EtcdConfig) (api.Client, error) {
 	}
 
 	// Create the etcd client
-	tlsInfo := &transport.TLSInfo{
-		CAFile:   config.EtcdCACertFile,
-		CertFile: config.EtcdCertFile,
-		KeyFile:  config.EtcdKeyFile,
+	// If Etcd Certificate and Key are provided inline through command line agrument,
+	// then the inline values take precedence over the once in the config file.
+	// All the three parametes, Certificate, key and CA certificate are to be provided inline for processing.
+	var tls *tls.Config
+	var err error
+	if (config.EtcdCert != "" && config.EtcdKey == "" && config.EtcdKeyFile != "") ||
+		(config.EtcdKey != "" && config.EtcdCert == "" && config.EtcdCertFile != "") ||
+		(config.EtcdCertFile != "" && config.EtcdKeyFile == "" && config.EtcdKey != "") ||
+		(config.EtcdKeyFile != "" && config.EtcdCertFile == "" && config.EtcdCert != "") {
+		return nil, fmt.Errorf("Cannot process mix of inline certificate-key and certificate files")
+	}
+
+	if config.EtcdCert != "" && config.EtcdKey != "" {
```

Comment:
Then, you can re-use the variables calculated above:
```
if haveInline {
   // We have inline etcd certs, the etcd TLSInfo doesn't support inline keys so we load the keys ourselves.
   ...
} else if haveFiles {
   ...
}
```

Reply:
Acknowledged and updated code.